app/rtc_s1.py

```diff
@@ -475,80 +669,163 @@ def run(cfg):
+                            abs_cal_factor=abs_cal_factor,
+                            flag_upsample_radar_grid=flag_upsample_radar_grid,
+                            clip_min = clip_min,
+                            clip_max = clip_max,
+                            # out_off_diag_terms=out_off_diag_terms_obj,
+                            out_geo_nlooks=out_geo_nlooks_obj,
+                            out_geo_rtc=out_geo_rtc_obj,
+                            input_rtc=None,
+                            output_rtc=None,
+                            dem_interp_method=dem_interp_method_enum,
+                            memory_mode=memory_mode)
+
+        if flag_error_sub_swaths:
+            logger.warning('WARNING the sub-swath masking is not available'
+                           ' from this ISCE3 version. The sub-swath masking'
+                           ' was disabled.')
+
+        # Calculate layover shadow mask when requested
+        if save_layover_shadow_mask:
+            layover_shadow_mask_file = (f'{bursts_output_dir}/{product_id}'
+               f'_layover_shadow_mask.{imagery_extension}')
+            calculate_layover_shadow_mask(burst,
+                                geogrid,
+                                cfg.dem,
+                                layover_shadow_mask_file,
+                                output_raster_format,
+                                threshold_rdr2geo=cfg.rdr2geo_params.threshold,
+                                numiter_rdr2geo=cfg.rdr2geo_params.numiter,
+                                threshold_geo2rdr=cfg.geo2rdr_params.threshold,
+                                numiter_geo2rdr=cfg.geo2rdr_params.numiter)
+            
+            if flag_bursts_files_are_temporary:
+                temp_files_list.append(layover_shadow_mask_file)
+            else:
+                output_file_list.append(layover_shadow_mask_file)
+                logger.info(f'file saved: {layover_shadow_mask_file}')
+            output_metadata_dict['layover_shadow_mask'][1].append(
+                layover_shadow_mask_file)
+
+        else:
+            layover_shadow_mask_file = None
 
         del geo_burst_raster
-        if not flag_bursts_files_are_temporary:
-            logger.info(f'file saved: {geo_burst_filename}')
+
+        # Output imagery list contains multi-band files that
+        # will be used for mosaicking
         output_imagery_list.append(geo_burst_filename)
 
-        if flag_save_nlooks:
+        # If burst imagery is not temporary, separate polarization channels
+        if not flag_bursts_files_are_temporary:
+            output_burst_imagery_list = []
+            for pol in pol_list:
+                geo_burst_pol_filename = \
+                    (f'{output_dir}/{burst_id}/{product_id}_{pol}.'
+                     f'{imagery_extension}')
+                output_burst_imagery_list.append(geo_burst_pol_filename)
+
+            _separate_pol_channels(geo_burst_filename,
```

Comment:
'_separate_pol_channels()` will save the image into geotiff format regardless of `imagery_extension`.